src/bodywork/workflow_execution.py

```diff
@@ -490,16 +506,63 @@ def parse_dockerhub_image_string(image_string: str) -> Tuple[str, str]:
     return image_name, image_tag
 
 
-def _print_logs_to_stdout(namespace: str, job_or_deployment_name: str) -> None:
+def _compute_optimal_job_timeout(batch_stages: List[BatchStageConfig]) -> int:
+    """Compute the optimal timeout for job monitoring.
+
+    :param namesapce: The target namespace for the job.
+    :param batch_stages: The desired configuration for the jobs.
+    :param returns: The optimal timeout (in seconds).
+    """
+    job_timeouts = [
+        max(1, stage.retries) * stage.max_completion_time for stage in batch_stages
+    ]
+    return int(max(job_timeouts) + TIMEOUT_GRACE_SECONDS)
+
+
+def _compute_optimal_deployment_timeout(
```

Comment:
I think these methods should be moved to `batch_jobs.py` and `deployments.py` respectively, as they refer to those specific things. Also `workflow_execution.py` is a pretty big file as it is.

What do you think?

Reply:
I thought it was a good idea at face value, but when I copied `_compute_optimal_job_timeout` into `bodywork.k8s.batch_jobs`, I realised that `BatchStageConfig` would need to be imported and that this isn't really a k8s entity - it's a Bodywork entity.

So I'm going to leave this out for the time being.

Reply:
That's a good point, it is more of a config entity.